hack/go-dev-linter.sh

```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env bash
+set -e
+
+source ./hack/common.sh
+
+fetch_go_linter
+
+##todo(camilamacedo86): It should be removed when we are able to enable all these linters checks in the hack/go-linter.sh
+header_text "Running dev golangci-lint"
+golangci-lint run --disable-all \
+    --deadline 5m \
+    --enable=nakedret \
+    --enable=golint \
+    --enable=gocyclo \
+    --enable=goimports \
+    --enable=lll \
+    --enable=goconst \
+    --enable=gosec \
+    --enable=maligned \
+    --enable=deadcode \
+    --enable=misspell \
+    --enable=errcheck \
+    --enable=dupl \
+    --enable=interfacer \
+    --enable=ineffassign \
+    --enable=varcheck \
+    --enable=structcheck \
+    --enable=unparam
+
```

Comment:
Rather than maintaining a separate script, how about we combine and store the CI and dev linters in separate array variables, and then use an argument to the script to choose which list.

```bash
#!/usr/bin/env bash
set -e

source ./hack/common.sh

CI_LINTERS=(
    "--enable=nakedret"
    "--enable=goconst"
)

DEV_LINTERS=(
    "--enable=nakedret"
    "--enable=golint"
    "--enable=gocyclo"
    "--enable=goimports"
    "--enable=lll"
    "--enable=goconst"
    "--enable=gosec"
    "--enable=maligned"
    "--enable=deadcode"
    "--enable=misspell"
    "--enable=errcheck"
    "--enable=dupl"
    "--enable=interfacer"
    "--enable=ineffassign"
    "--enable=varcheck"
    "--enable=structcheck"
    "--enable=unparam"
)

subcommand=$1
case $subcommand in
	"dev")
		LINTERS=${DEV_LINTERS[@]}
		;;
	"ci")
		LINTERS=${CI_LINTERS[@]}
		;;
	*)
		echo "Must pass 'dev' or 'ci' argument"
		exit 1
esac

fetch_go_linter

header_text "Running golangci-lint"
golangci-lint run --disable-all \
    --deadline 5m ${LINTERS[@]}
```

Reply:
I did not want to put all together because I thought that the goal is we be able to remove this one as soon as possible.